source/docs/software/wpilib-tools/robotbuilder/writing-code/robotbuilder-writing-subsystem-code.rst

```diff
@@ -19,56 +17,113 @@ Adding Subsystem Capabilities
+
+          }
+
+          public void open() {
+              motor.set(1.0);
+          }
+
+          public void close() {
+              motor.set(-1.0);
+          }
+
+          public void stop() {
+              motor.set(0.0);
+          }
+
+          public boolean isGripping() {
+              return limitswitch.get();
+          }
 
-         public void stop() {
-            victor.set(0.0);
-         }
       }
 
    .. code-tab:: cpp
 
-      Claw::Claw() :
-         Subsystem("Claw") {
-         // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
-         victor = RobotMap::CLAW_VICTOR;
-         // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+      Claw::Claw(){
+      SetName("Claw");
+          // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+       AddChild("motor", &m_motor);
+       m_motor.SetInverted(false);
+
+       AddChild("limit switch", &m_limitswitch);
+
+
+          // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+      }
+
+      void Claw::Periodic() {
+          // Put code here to be run every loop
+
       }
 
-      void Claw::InitDefaultCommand() {
-         // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DEFAULT_COMMAND
-         // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DEFAULT_COMMAND
+      void Claw::SimulationPeriodic() {
+          // This method will be called once per scheduler run when in simulation
+
       }
 
+      // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CMDPIDGETTERS
+
+      // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CMDPIDGETTERS
+
       void Claw::Open() {
-         victor->Set(1);
+          m_motor.Set(1.0);
       }
 
       void Claw::Close() {
-         victor->Set(-1);
+          m_motor.Set(-1.0);
       }
 
       void Claw::Stop() {
-         victor->Set(0);
+          m_motor.Set(0.0);
+      }
+
+      bool Claw::IsGripping() {
+          return m_limitswitch.Get();
       }
 
-Add methods to the claw.java that will open, close, and stop the claw from moving. Those will be used by commands that actually operate the claw. The comments have been removed from this file to make it easier to see the changes for this document. Notice that a member variable called "victor" is created by RobotBuilder so it can be used throughout the subsystem. Each of your dragged-in palette items will have a member variable with the name given in RobotBuilder.
+Add methods to the claw.java/cpp that will open, close, and stop the claw from moving and get the claw limit switch. Those will be used by commands that actually operate the claw.
+
+.. note:: The comments have been removed from this file to make it easier to see the changes for this document.
+
+Notice that member variable called "motor" and "limitswitch" are created by RobotBuilder so it can be used throughout the subsystem. Each of your dragged-in palette items will have a member variable with the name given in RobotBuilder.
```

Comment:
```suggestion
Notice that member variable called ``motor`` and ``limitswitch`` are created by RobotBuilder so it can be used throughout the subsystem. Each of your dragged-in palette items will have a member variable with the name given in RobotBuilder.
```